File: CodeQuestAcademy/src/AnagramChecker.java
import java.util.Arrays;
import java.util.Scanner;
// Problem 28: Anagram Checker
// Originally Published: Code Quest 2016
// Difficulty: Easy
// Code Written By: Sam L.
public class AnagramChecker {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int T = sc.nextInt();
        for (int i = 0; i < T; i++)
        {
            String str = sc.next();
            String[] s = str.split("\\|");
            char[] s1 = s[0].toCharArray();
            char[] s2 = s[1].toCharArray();
            Arrays.sort(s1);
            Arrays.sort(s2);
            System.out.print(str + " = ");
            if (Arrays.equals(s1, s2) && !s[0].equals(s[1]))
                System.out.println("ANAGRAM");
            else
                System.out.println("NOT AN ANAGRAM");
        }
    }
}

File: CodeQuestAcademyC++/Easy/AnagramChecker.cpp
/*
Problem 28: Anagram Checker
Difficulty: Easy
Originally Published: Code Quest 2016
*/

#include <bits/stdc++.h>
#define lli long long int
using namespace std; 

lli T;
string S, S1, S2, S3, S4;

int main()
{
	cin >> T;
	for (lli t = 0; t < T; t++) {
		cin >> S;
		// Get String inputs
		S1 = S.substr(0, S.find("|"));
		S2 = S.substr(S.find("|") + 1);
		S3 = S1;
		S4 = S2;
		sort(S1.begin(), S1.end());
		sort(S2.begin(), S2.end());
		// Check for anagram
		if (S1 == S2 && S3 != S4) {
			cout << S << " = ANAGRAM" << endl;
		} else {
			cout << S << " = NOT AN ANAGRAM" << endl;
		}
	}
	return 0;
}

/*
Sample Input:
11
STOP|POTS
ADMIRER|MARRIED
CAT|DOG
CREATIVE|REACTIVE
LISTEN|SILENT
ANGERED|ENRAGED
ELVIS|LIVES
RUN|FLY
DEDUCTIONS|DISCOUNTED
PATERNAL|PARENTAL
MIKE|MIKE
*/

/*
Sample Output:
STOP|POTS = ANAGRAM
ADMIRER|MARRIED = ANAGRAM
CAT|DOG = NOT AN ANAGRAM
CREATIVE|REACTIVE = ANAGRAM
LISTEN|SILENT = ANAGRAM
ANGERED|ENRAGED = ANAGRAM
ELVIS|LIVES = ANAGRAM
RUN|FLY = NOT AN ANAGRAM
DEDUCTIONS|DISCOUNTED = ANAGRAM
PATERNAL|PARENTAL = ANAGRAM
MIKE|MIKE = NOT AN ANAGRAM
*/